kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelHandler.java

```diff
@@ -331,38 +223,46 @@ private CompletableFuture<ChannelHandlerContext> authenticateInternal(ChannelHan
                     saslAuthBytes = new byte[0];
                     break;
             }
-            SaslAuthenticateRequest request = new SaslAuthenticateRequest
-                    .Builder(ByteBuffer.wrap(saslAuthBytes))
-                    .build();
 
-            ByteBuffer buffer = request.serialize(header);
-
-            KafkaCommandDecoder.KafkaHeaderAndRequest fullRequest = new KafkaCommandDecoder.KafkaHeaderAndRequest(
-                    header,
-                    request,
-                    Unpooled.wrappedBuffer(buffer),
-                    null
+            final PendingRequest pendingRequest = new PendingRequest(
+                    ApiKeys.SASL_AUTHENTICATE,
+                    correlationIdGenerator.next(),
+                    newSaslAuthenticate(saslAuthBytes),
+                    __ -> {}
             );
-            sendGenericRequestOnTheWire(channel, fullRequest, result);
-
+            enqueueRequest(channel, pendingRequest);
```

Comment:
@BewareMyPower Not sure why, after cherry-picking this PR, branch-2.9 will lose this line. I have re-cherry-picked this PR and added this line. The test will not fail now.

https://github.com/streamnative/kop/blob/e298ccb803d1bc132c079e31799f7720a9616265/kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelHandler.java#L226-L234